crates/bevy_ptr/src/lib.rs

```diff
@@ -290,3 +290,95 @@ impl<'a, T> UnsafeCellDeref<'a, T> for &'a UnsafeCell<T> {
         self.get().read()
     }
 }
+
+/// A type similar to [`UnsafeCell`] except it can be constructed from a shared or mutable reference.
+///
+/// This type tries to act "borrow-like" like which means that:
+/// - It must always reference a valid instance of `T`.
+/// - The lifetime `'a` accurately represents how long it is valid.
+///
+/// However, while this cell can produce a mutable reference (only if created from one),
+/// it *can* be cloned, so the user is responsible for upholding Rust's aliasing rules.
+/// - A `&T` reference may be released to safe code and may co-exist with other `&T` references,
+/// but not with a `&mut T` reference.
+/// - A `&mut T` reference may be released to safe code provided no other `&mut T` or `&T` exists. A `&mut T` must always be unique.
+///
+#[doc(hidden)]
```

Comment:
Why #[doc(hidden)] ?

Reply:
I don't remember why I did that. I'll remove it.

Reply:
```suggestion
```